omod-1.8/src/main/java/org/openmrs/module/webservices/rest/web/v1_0/resource/openmrs1_8/TaskActionResource1_8.java

```diff
@@ -0,0 +1,220 @@
+	}
+	
+	// Stop and start a scheduled task.
+	private void reScheduleTask(Collection<TaskDefinition> taskDefs) {
+		for (TaskDefinition taskDef : taskDefs) {
+			try {
+				taskServiceWrapper.reScheduleTask(taskDef);
+			}
+			catch (SchedulerException e) {
+				throw new APIException("Errors occurred while rescheduling task", e);
+			}
+		}
+	}
+	
+	// Stop and start a scheduled all task.
+	private void reScheduleAllTask() {
+		try {
+			taskServiceWrapper.reScheduleAllTasks();
+		}
+		catch (SchedulerException e) {
+			throw new APIException("Errors occurred while rescheduling all tasks", e);
+		}
+		
+	}
+	
+	// Start all tasks that are scheduled to run on startup.
+	private void onStartUpTask() {
+		taskServiceWrapper.onStartUpTask();
+	}
+	
+	// Stop all tasks and clean up the scheduler instance.
+	private void onShutDownUpTask() {
+		taskServiceWrapper.onShutDownUpTask();
+	}
+	
+	private void deleteTask(Collection<TaskDefinition> taskDefs) {
+		for (TaskDefinition taskDef : taskDefs) {
+			try {
+				taskServiceWrapper.unloadTask(taskDef);
+			}
+			catch (SchedulerException e) {
+				throw new APIException("Errors occurred while deleting task", e);
+			}
+		}
+	}
+	
+	@Override
+	public TaskAction newDelegate() {
+		return new TaskAction();
+	}
+	
+	@Override
+	public TaskAction save(TaskAction delegate) {
+		throw new UnsupportedOperationException("TaskAction cannot be saved");
+	}
+	
+	@Override
+	public DelegatingResourceDescription getRepresentationDescription(Representation rep) {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("tasks", Representation.REF);
+		description.addProperty("action", "action");
+		return description;
+	}
+	
+	@Override
+	public TaskAction getByUniqueId(String uniqueId) {
+		throw new UnsupportedOperationException("ModuleAction cannot get by id");
+	}
+	
+	@Override
+	protected void delete(TaskAction delegate, String reason, RequestContext context) throws ResponseException {
+		throw new UnsupportedOperationException("ModuleAction cannot be deleted");
+	}
+	
+	@Override
+	public void purge(TaskAction delegate, RequestContext context) throws ResponseException {
+		throw new UnsupportedOperationException("ModuleAction cannot be purged");
+	}
+	
+	@Override
+	public DelegatingResourceDescription getCreatableProperties() throws ResourceDoesNotSupportOperationException {
+		DelegatingResourceDescription description = new DelegatingResourceDescription();
+		description.addProperty("tasks");
+		description.addProperty("allTasks");
+		description.addRequiredProperty("action", "action");
+		return description;
+	}
+	
+	/**
+	 * Converter does not handle getters starting with 'is' instead of 'get'
+	 */
+	@PropertyGetter("allModules")
+	public Boolean isAllModules(TaskAction action) {
+		return action.isAllTasks();
+	}
+	
+	private ServletContext getServletContext(RequestContext context) {
+		return context.getRequest().getSession().getServletContext();
```

Comment:
Can you remind me what this servlet context is used for?

Reply:
Removed, It's not required here :-)